pkg/component/controller/coredns.go

```diff
@@ -286,8 +286,8 @@ func (c *CoreDNS) Init() error {
 }
 
 // Run runs the CoreDNS reconciler component
-func (c *CoreDNS) Run() error {
-	c.stopCtx, c.stopFunc = context.WithCancel(context.Background())
+func (c *CoreDNS) Run(ctx context.Context) error {
+	c.stopCtx, c.stopFunc = context.WithCancel(ctx)
```

Comment:
why is it a field, is it used outside of the Run function anywhere? if no, can be a local variable

Reply:
`c.stopFunc` is called in the `Stop()` function:
https://github.com/k0sproject/k0s/blob/be438c564d58ed81431e8d11b5ff25605e06e196/pkg/component/controller/coredns.go#L353

Reply:
Done